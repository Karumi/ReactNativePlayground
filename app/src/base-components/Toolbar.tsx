import {  Body, Header, Title } from "native-base";
import * as React from "react";
import translator from "../i18n";

class Toolbar extends React.Component {
    public render() {
        return (
            <Header>
                <Body>
                    <Title>{translator().appName}</Title>
                </Body>
            </Header>
        );
    }
}

export default Toolbar;
