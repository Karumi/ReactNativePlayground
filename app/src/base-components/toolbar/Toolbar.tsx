import { Body, Button, Header, Left, Title } from "native-base";
import * as React from "react";
import { Image } from "react-native";
import translator from "../../i18n";
import { ios } from "../../platform";
import assets from "./images/assets";

export interface Props {
    navigation: any;
    showBackButton?: boolean;
    title?: string;
}

class Toolbar extends React.Component<Props> {

    public static defaultProps: Partial<Props> = {
        showBackButton: false,
        title: translator().appName,
    };

    public render() {
        const showBack = this.props.showBackButton;
        const titleLeftPadding = ios() && showBack ? -12 : 0;
        const leftItem = showBack ? this.backButton() : null;
        return (
            <Header>
                <Left style={{ flex: 0, paddingLeft: 0 }} >
                    {leftItem}
                </Left>
                <Body style={{ left: titleLeftPadding }}>
                    <Title>{this.props.title}</Title>
                </Body>
            </Header>
        );
    }
    private backButton() {
        return (
            <Button transparent onPress={() => this.props.navigation.pop()}>
                <Image source={assets.arrowBack} />
            </Button>
        );
    }
}

export default Toolbar;
