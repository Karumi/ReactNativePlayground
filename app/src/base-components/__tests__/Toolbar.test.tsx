
import * as React from "react";
import * as renderer from "react-test-renderer";
import Toolbar from "../Toolbar";

it("renders correctly with defaults", () => {
    const toolbar = renderer.create(<Toolbar />).toJSON();
    expect(toolbar).toMatchSnapshot();
});
