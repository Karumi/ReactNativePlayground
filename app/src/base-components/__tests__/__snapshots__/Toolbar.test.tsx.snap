// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with defaults 1`] = `
<View
  onLayout={[Function]}
>
  <View
    style={
      Object {
        "backgroundColor": "#F8F8F8",
        "borderBottomColor": "#a7a6ab",
        "borderBottomWidth": 0.5,
        "elevation": 3,
        "flexDirection": "row",
        "height": 64,
        "justifyContent": "center",
        "left": 0,
        "paddingLeft": 6,
        "paddingRight": 10,
        "paddingTop": 18,
        "right": 0,
        "shadowColor": undefined,
        "shadowOffset": undefined,
        "shadowOpacity": undefined,
        "shadowRadius": undefined,
        "top": 0,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "flex": 1,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 17,
            "fontWeight": "700",
            "marginLeft": undefined,
            "paddingLeft": 4,
            "paddingTop": 1,
            "textAlign": "center",
          }
        }
      >
        React Native Playground
      </Text>
    </View>
  </View>
</View>
`;
